### PR TITLE
Add helpers to create Docker agents in integration tests

### PR DESCRIPTION
**Pull request by uhafner**
Adds a `JavaGitContainer` that can be used as an agent in integration tests.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -130,6 +130,18 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <!-- Docker Agents -->
+    <dependency>
+      <groupId>org.jenkins-ci.test</groupId>
+      <artifactId>docker-fixtures</artifactId>
```
Comment (timja) [inline pom.xml:136]: This is deprecated, testcontainers is generally used instead

Comment (uhafner) [reply pom.xml:136]: Do we have an example?

Comment (timja) [reply pom.xml:136]: Lots of general examples:
https://github.com/search?q=org%3Ajenkinsci+testcontainers&type=code

or
https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Ajenkinsci+testcontainers

Comment (uhafner) [reply pom.xml:136]: Ah, it's https://www.testcontainers.org. I thought a new Jenkins approach. Thanks for the pointers!

**Pull request closed without merging**